### PR TITLE
[#10285] IRCClient: Split NOTICEs the same way as PRIVMSGs before sending

### PR DESCRIPTION
**Pull request by DefaultUser**
## Scope and purpose

NOTICEs should follow the same length restrictions as PRIVMSGs.
Additionally they should also be automatically split on newline
characters to have a consistent interface with "msg".

## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10285
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] The title of the PR starts with the associated Trac ticket number (without the `#` character).
* [x] I have updated the automated tests and checked that all checks for the PR are green.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.
* [x] The merge commit will use the below format
  The first line is automatically generated by GitHub based on PR ID and branch name.
  The other lines generated by GitHub should be replaced.

```
Merge pull request #123 from twisted/4356-branch-name-with-trac-id

Author: DefaultUser
Reviewer:
Fixes: ticket: 10285
```

### CODE REVIEW

File: src/twisted/words/protocols/irc.py
Comment (twm) [inline src/twisted/words/protocols/irc.py]: As this enum isn't used by any public API surface it should have a private name.

I think it'd be fine &mdash; even preferable &mdash; to pass the strings `'PRIVMSG'` and `'NOTICE'` directly without the enum at all, again because it's all private. Enums are slow after all.

File: src/twisted/words/protocols/irc.py
Comment (twm) [inline src/twisted/words/protocols/irc.py]: By the Twisted naming convention this should be `_sendMsgOrNotice`. The same applies to the `msg_or_notice` argument &mdash; `msgOrNotice` &mdash; though the naming is awkward in both cases. I see the IRC3 spec refers to PRIVMSG and NOTICE as "messages". Perhaps call the function `_sendMessage` and the argument `command`?

File: src/twisted/words/test/test_irc.py
Comment (twm) [inline src/twisted/words/test/test_irc.py]: Please split these tests apart so that each one tests a single thing: `msg` or `notice`, or convert them to be table driven. The former is probably easier. As they are now constructed they're two tests glommed together, but we want to test the behavior of the API surface separately.

**Review by twm — CHANGES_REQUESTED** [review]
Hi, thank you for giving IRCClient some love! The mechanics of this look good, but there are some minor changes to be made before merge. Thanks for your persistence with the process!

File: src/twisted/words/protocols/irc.py
Comment (DefaultUser) [reply src/twisted/words/protocols/irc.py]: I wanted to avoid ambiguous names, but if the specs refer to both as "messages", it should be allright.

File: src/twisted/words/protocols/irc.py
Comment (DefaultUser) [reply src/twisted/words/protocols/irc.py]: I prefer passing enums over string constants as it's easier to spot typos that way, but I guess that's not a prblem here as the usage is very limited.

**Review by twm — APPROVED** [review]
Looks good, and thanks again! I'll merge this once the builds finish.

**Pull request merged**
